plugin/src/main/java/com/playmonumenta/scriptedquests/zones/zone/ZoneFragment.java

```diff
@@ -15,17 +17,28 @@
  */
 public class ZoneFragment<T> extends BaseZone {
 	private HashMap<String, Zone<T>> mParents = new HashMap<String, Zone<T>>();
+	private HashMap<String, ArrayList<Zone<T>>> mParentsAndEclipsed = new HashMap<String, ArrayList<Zone<T>>>();
 	private boolean mValid;
 
 	public ZoneFragment(ZoneFragment<T> other) {
 		super(other);
 		mParents.putAll(other.mParents);
+		for (Map.Entry<String, ArrayList<Zone<T>>> entry : other.mParentsAndEclipsed.entrySet()) {
+			String layerName = entry.getKey();
+			ArrayList<Zone<T>> otherZones = entry.getValue();
+
+			ArrayList<Zone<T>> zones = new ArrayList<Zone<T>>();
+			zones.addAll(otherZones);
+			mParentsAndEclipsed.put(layerName, zones);
```

Comment:
All the list types have more efficient copy constructors - so you can do: 
```
mParentsAndEclipsed.put(entry.getValue(), new ArrayList<Zone<T>>(entry.getValue()));
```

Which will get a new list that's appropriately sized for the input data